astroplan/scheduling.py

```diff
@@ -277,36 +439,35 @@ def _make_schedule(self, blocks):
 
             if block_constraint_results[bestblock_idx] == 0.:
                 # if even the best is unobservable, we need a gap
-                new_blocks.append(TransitionBlock({'nothing_observable': self.gap_time},
-                                                  current_time))
                 current_time += self.gap_time
             else:
                 # If there's a best one that's observable, first get its transition
                 trans = block_transitions.pop(bestblock_idx)
                 if trans is not None:
-                    new_blocks.append(trans)
+                    self.schedule.insert_slot(trans.start_time, trans)
```

Comment:
These will need to be inside ```try...except``` blocks, and you'll have to remember to remove the transition block if the main observing block cannot be inserted.

In fact, I wonder if this should be a loop over the blocks in order of priority, which inserts the first block which is insertable at the ```current_time```?